### PR TITLE
Implement dummy approved languages table

### PR DESCRIPTION
**Pull request by Puepis**
## Notion ticket link

https://www.notion.so/uwblueprintexecs/Create-dummy-Approved-Languages-Levels-table-5de91af7e3064d70b10e6dade6133b6b

## Implementation description

- Created new table component to display user's Approved Languages & Levels
- Created a dummy slider

image

## Steps to test

1. Login to angela merkel and go to any `/user/:id`
2. Ensure that the table looks correct

## What should reviewers focus on?

- Does the slider look okay?
- Does the sorting work?

## Checklist

- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] For backend changes, I have run the appropriate linters: `docker exec -it planet-read_py-backend_1 /bin/bash -c "black . && isort --profile black ."` and I have generated new migrations: `flask db migrate -m "<your message>"`
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/admin/ApprovedLanguagesTable.tsx [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+      setIsAscending: setIsAscendingRole,
+    },
+  };
+
+  const sortApprovedLanguages = (field: string) => {
+    const newApprovedLanguages = [...approvedLanguages];
+    const { sortFn, isAscending, setIsAscending } = fieldSortDict[field];
+    setIsAscending(!isAscending);
+    newApprovedLanguages.sort(sortFn);
+    console.log(newApprovedLanguages);
```
Comment (gaoxk) [inline frontend/src/components/admin/ApprovedLanguagesTable.tsx:67]: nit: remove console log

File: frontend/src/components/admin/ApprovedLanguagesTable.tsx [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+  };
+
+  useEffect(() => {
+    if (approvedLanguagesTranslation && approvedLanguagesReview) {
+      const translationLanguages = Object.entries(
+        approvedLanguagesTranslation,
+      ).map(([language, level]) => {
+        return { language, level, role: "Translator" };
+      });
+      const reviewLanguages = Object.entries(approvedLanguagesTranslation).map(
```
Comment (gaoxk) [inline frontend/src/components/admin/ApprovedLanguagesTable.tsx:78]: little bug: `reviewLanguages` is using `approvedLanguagesTranslation`. Eg carl sagan's table (he has no approved review languages)
image

File: frontend/src/components/admin/ApprovedLanguagesTable.tsx [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+        borderBottom={
+          index === approvedLanguages.length - 1 ? "1px solid gray" : ""
+        }
+        key={`${approvedLanguage.language}/${approvedLanguage.role}`}
+      >
+        <Td>{convertLanguageTitleCase(approvedLanguage.language)}</Td>
+        <Td>{approvedLanguage.role}</Td>
+        <Td colSpan={4} align="center">
+          <Slider
+            defaultValue={1}
```
Comment (gaoxk) [inline frontend/src/components/admin/ApprovedLanguagesTable.tsx:101]: nit: what happens if you do `defaultValue={approvedLanguage.level}`

**Review by gaoxk — CHANGES_REQUESTED** [review]
very nice 😎

File: frontend/src/components/admin/ApprovedLanguagesTable.tsx (hunk 2 above)
Comment (Puepis) [reply frontend/src/components/admin/ApprovedLanguagesTable.tsx:78]: shoot mb, will fix that

File: frontend/src/components/admin/ApprovedLanguagesTable.tsx (hunk 3 above)
Comment (Puepis) [reply frontend/src/components/admin/ApprovedLanguagesTable.tsx]: i'll add an additional field called `sliderValue` to use instead (will help for making the slider functional)

**Comment by Puepis** [comment]
fixed a merge conflict and now both tables show on the user profile page (and both have slightly different styling), how should I make it consistent?

image

File: frontend/src/components/admin/ApprovedLanguagesTable.tsx [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+          return { language, level, role: "Reviewer", sliderValue: level };
+        },
+      );
+      const combined = translationLanguages.concat(reviewLanguages);
+      sortApprovedLanguages("language");
+      setApprovedLanguages(combined);
+    }
+  }, [approvedLanguagesTranslation, approvedLanguagesReview]);
+
+  const onSliderValueChange = (index: number, newValue: number) => {
```
Comment (gaoxk) [inline frontend/src/components/admin/ApprovedLanguagesTable.tsx:90]: nit: add a todo comment along the lines of `// TODO: call gql mutation to update new level`

File: frontend/src/components/admin/ApprovedLanguagesTable.tsx [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+            </SliderTrack>
+            <SliderThumb />
+            {[1, 2, 3, 4].map((val) => (
+              <SliderMark
+                key={val}
+                value={val}
+                height="6px"
+                width="4px"
+                marginTop="-3px"
+                bg={approvedLanguage.sliderValue > val ? "#64A1CD" : "#ECF1F4"}
```
Comment (gaoxk) [inline frontend/src/components/admin/ApprovedLanguagesTable.tsx:128]: hmmm can chakraui colour codes work here?

**Review by gaoxk — CHANGES_REQUESTED** [review]
> fixed a merge conflict and now both tables show on the user profile page (and both have slightly different styling), how should I make it consistent?

found the difference! In the `AssignedStoryTranslationsTable.tsx` remove `size="sm"` on line 129

**Review by gaoxk — APPROVED** [review]

**Pull request merged**